aptos-move/framework/aptos-stdlib/sources/cryptography/bls12381.move

```diff
@@ -0,0 +1,661 @@
+    ///
+    /// Return `true` if the BLS `multisignature` on `message` verifies against the BLS aggregate public key `agg_public_key`.
+    /// Returns `false` otherwise.
+    /// Does not abort.
+    native fun verify_multisignature_internal(
+        multisignature: vector<u8>,
+        agg_public_key: vector<u8>,
+        message: vector<u8>
+    ): bool;
+
+    /// CRYPTOGRAPHY WARNING: This function WILL check that the public key is a prime-order point, in order to prevent
+    /// library users from misusing the library by forgetting to validate public keys before giving them as arguments to
+    /// this function.
+    ///
+    /// Returns `true` if the `signature` on `message` verifies under `public key`.
+    /// Returns `false` otherwise.
+    /// Does not abort.
+    native fun verify_normal_signature_internal(
+        signature: vector<u8>,
+        public_key: vector<u8>,
+        message: vector<u8>
+    ): bool;
+
+    /// Return `true` if the bytes in `public_key` are a valid bls12381 public key (as per `validate_pubkey`)
+    /// *and* this public key has a valid proof-of-possesion (PoP).
+    /// Return `false` otherwise.
+    /// Does not abort.
+    native fun verify_proof_of_possession_internal(
+        public_key: vector<u8>,
+        proof_of_possesion: vector<u8>
+    ): bool;
+    spec verify_proof_of_possession { // TODO: temporary mockup.
+        pragma opaque;
+    }
+
+    /// CRYPTOGRAPHY WARNING: Assumes the public key has a valid proof-of-possesion (PoP). This prevents rogue-key
+    /// attacks later on during signature aggregation.
+    ///
+    /// Returns `true` if the `signature_share` on `message` verifies under `public key`.
+    /// Returns `false` otherwise, similar to `verify_multisignature`.
+    /// Does not abort.
+    native fun verify_signature_share_internal(
+        signature_share: vector<u8>,
+        public_key: vector<u8>,
+        message: vector<u8>
+    ): bool;
+
+    //
+    // Constants and helpers for tests
+    //
+
+    /// Random signature generated by running `cargo test -- bls12381_sample_signature --nocapture --include-ignored` in `crates/aptos-crypto`.
+    /// The message signed is "Hello Aptos!" and the associated SK is 07416693b6b32c84abe45578728e2379f525729e5b94762435a31e65ecc728da.
+    const RANDOM_SIGNATURE: vector<u8> = x"a01a65854f987d3434149b7f08f70730e30b241984e8712bc2aca885d632aafced4c3f661209debb6b1c8601326623cc16ca2f6c9edc53b7b88b7435fb6b05ddece418d2c34dc6aca2f5a11a79e67774582c14084a01dcb7820e4cb4bad0ea8d";
+
+    /// Random signature generated by running `cargo test -- bls12381_sample_signature --nocapture --include-ignored` in `crates/aptos-crypto`.
+    /// The associated SK is 07416693b6b32c84abe45578728e2379f525729e5b94762435a31e65ecc728da.
+    const RANDOM_PK: vector<u8> = x"8a53e7ae5270e3e765cd8a4032c2e77c6f7e87a44ebb85bf28a4d7865565698f975346714262f9e47c6f3e0d5d951660";
+
+    fun get_random_aggsig(): AggrOrMultiSignature {
```

Comment:
add `#[test_only]` here and to any others, that makes auditing code a lot easier.